File: Flyweight/Oval.cs
namespace Flyweight
{
    internal class Oval : IShape
    {
        private readonly int x;
        private readonly int y;
        private readonly int width;
        private readonly int height;
        private readonly Color color;
        private readonly bool fill;

        public Oval(int x, int y, int width, int height, Color color, bool fill)
        {
            this.x = x;
            this.y = y;
            this.width = width;
            this.height = height;
            this.color = color;
            this.fill = fill;
        }

        public void Draw(Graphics g)
        {
            DrawingForm.Instance.Invoke((MethodInvoker)(() => DrawingForm.Instance.ClearMessage()));
            DrawingForm.Instance.Invoke((MethodInvoker)(() => DrawingForm.Instance.ShowMessage("Creating Oval...")));
            if (fill)
                g.FillEllipse(new SolidBrush(color), x, y, width, height);
            else
                g.DrawEllipse(new Pen(color), x, y, width, height);
        }
    }
}

File: Java/Flyweight/src/Oval.java
import java.awt.*;

class Oval implements IShape {
    private final int x;
    private final int y;
    private final int width;
    private final int height;
    private final Color color;
    private final boolean fill;

    public Oval(int x, int y, int width, int height, Color color, boolean fill) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
        this.color = color;
        this.fill = fill;
    }

    @Override
    public void draw(Graphics g) {
        DrawingForm.Instance.clearMessage();
        DrawingForm.Instance.showMessage("Creating Oval...");
        g.setColor(color);
        if (fill)
            g.fillOval(x, y, width, height);
        else
            g.drawOval(x, y, width, height);
    }
}